pkg/receive/handler.go

```diff
@@ -49,16 +50,42 @@ var (
 		},
 		[]string{"handler"},
 	)
+	replicationRequestsTotal = prometheus.NewCounter(
```

Comment:
let's move not only the registration, but also the initialization of these to the `NewHandler` function

Reply:
Then we should also pass the `prometheus.Registry`?!

Reply:
Precisely.

Reply:
The registry is already passed into NewHandler via the options struct :+1: